src/util/codec/bytes.rs

```diff
@@ -55,13 +58,26 @@ pub trait BytesEncoder: NumberEncoder {
     }
 }
 
+fn adjust_bytes_order<'a>(bs: &'a [u8], desc: bool, buf: &'a mut [u8]) -> &'a [u8] {
+    if desc {
+        let mut buf_idx = 0;
+        for &b in bs {
+            buf[buf_idx] = !b;
+            buf_idx += 1;
+        }
+        &buf[..buf_idx]
+    } else {
+        bs
+    }
+}
+
 impl<T: Write> BytesEncoder for T {}
 
 // Refer: https://github.com/facebook/mysql-5.6/wiki/MyRocks-record-format#memcomparable-format
-pub fn encode_bytes(key: &[u8]) -> Vec<u8> {
+pub fn encode_bytes(key: &[u8], desc: bool) -> Vec<u8> {
```

Comment:
Why not supplying another decode_bytes_desc?
Btw, I think using asc as the default argument seems more common.